lib/images.js

```diff
@@ -18,61 +18,75 @@
 const gcloud = require('google-cloud');
 const config = require('../config/config');
 const sharp = require('sharp');
-
+const strongDataUri = require('strong-data-uri');
+const stream = require('stream');
 const CLOUD_BUCKET = config.get('CLOUD_BUCKET');
-
+const dataFetcher = require('../lib/data-fetcher');
 const storage = gcloud.storage({
   projectId: config.get('GCLOUD_PROJECT')
 });
 const bucket = storage.bucket(CLOUD_BUCKET);
-const fetch = require('node-fetch');
 const CACHE_CONTROL_EXPIRES = 60 * 60 * 24; // 1 day.
 
 /**
- * Fetches and Saves an Image to GCloud.
+ * Fetches and Saves an Image to Google Cloud Storage.
  *
  * @param {string} url image URL to retrieve
- * @param {string} destFile name of the destination file on GCP
- * @return {Promise<string>}
+ * @param {string} destFile name of the destination file
+ * @return {Promise<url[]>>} URLs for the new images in Google Cloud Storage
  */
 function fetchAndSave(url, destFile) {
-  return new Promise((resolve, reject) => {
-    fetch(url)
-      .then(response => {
-        if (response.status !== 200) {
-          reject(
-            new Error('Bad Response (' + response.status + ') loading image: ' + response.url));
-        }
-        const contentType = response.headers.get('Content-Type');
+  if (url.startsWith('data:')) {
```

Comment:
So, we currently support `http`, `https` and `data` protocol. We could write this as  
switch(protocol) {
  case 'data': {
    //...
  } 
  case 'http': 
  case 'https': {
    // ...
  }
  default: {
    return Promise.reject('Unsupported Protocol: ' + protocol);
  }
}